rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -0,0 +1,234 @@
+from icon_rapid7_insight_agent.util.graphql_api.region_map import region_map
+from insightconnect_plugin_runtime.exceptions import PluginException, ConnectionTestException
+import icon_rapid7_insight_agent.util.graphql_api.agent_typer as agent_typer
+import requests
+
+class ApiConnection():
+    def __init__(self, api_key, org_key, region_string, logger):
+        self.api_key = api_key
+        self.org_key = org_key
+        self.logger = logger
+
+        self.endpoint = self._setup_endpoint(region_string)
+
+        self.session = requests.Session()
+        self.session.headers = self._get_headers()
+
+    def get_agent(self, agent_input):
+        agent_type = agent_typer.get_agent_type(agent_input)
+        agents = self._get_all_agents()
+        agent = self._find_agent_in_agents(agents, agent_input, agent_type)
+        return agent
+
+    def quarantine(self, advertisement_period, agent_id, quarantine_state):
+        """
+        Take a quarantine action on a given agent id
+
+        :param advertisement_period: int (Amount of time in seconds to try to take the quarantine action)
+        :param agent_id: string
+        :param quarantine_state: boolean (quarantine or unquarantine the agent)
+
+        :return: boolean
+        """
+        if quarantine_state:
+            quarantine_payload = {
+                "query": "mutation( $orgID:String! $agentID:String! $advPeriod:Long! ) { quarantineAssets( orgId:$orgID assetIds: [$agentID] input: {advertisementPeriod: $advPeriod} ) { results { assetId failed } } }",
+                "variables": {
+                    "orgID": self.org_key,
+                    "agentID": agent_id,
+                    "advPeriod": advertisement_period
+                }
+            }
+        else:
+            quarantine_payload = {
+                "query": "mutation( $orgID:String! $agentID:String!) { unquarantineAssets( orgId:$orgID assetIds: [$agentID] ) { results { assetId failed } } }",
+                "variables": {
+                    "orgID": self.org_key,
+                    "agentID": agent_id
+                }
+            }
+
+        action_verb = "quarantine" if quarantine_state else "unquarantine"
+        self.logger.info(f"Attempting to {action_verb} asset {agent_id} at {self.endpoint}")
+
+        results_object = self._post_payload(quarantine_payload)
+
+        if quarantine_state:
+            failed = results_object.get("data").get("quarantineAssets").get("results")[0].get("failed")
+        else:
+            failed = results_object.get("data").get("unquarantineAssets").get("results")[0].get("failed")
+
+        return (not failed)
+
+    def run_connection_test(self):
```

Comment:
```suggestion
    def test_connection(self):
```
Think about the API code less in terms of the plugin, and more in terms of general use